### PR TITLE
NewRule: SpecFocus

### LINKED ISSUE #172: [Feature Request] Detect focus in specs
It would be cool if it detected `focus: true` hanging around in specs.

```crystal
describe "Some specs I've been working on ", focus: true do
  #...
end
```

This could be useful for if you have a CI that auto deploys your app once specs/formatting/ameba all pass. You might be working on a large feature and had specs focused, but forgot to take it off before committing and pushing up. Specs pass, but you may have broke other specs that didn't actually run from `crystal spec`:grimacing: (not that I'm speaking from experience or anything)... In this "example", ameba would fail and app would not be deployed.

### PR DESCRIPTION
**Pull request by veelenga**
closes #172

image

### CODE REVIEW

File: src/ameba/rule/lint/spec_focus.cr [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    SPEC_ITEM_NAMES = %w(describe context it pending)
+
+    def test(source)
+      return unless source.path.ends_with?("_spec.cr")
+
+      AST::NodeVisitor.new self, source
+    end
+
+    def test(source, node : Crystal::Call)
+      return unless SPEC_ITEM_NAMES.includes?(node.name)
```
Comment (Sija) [inline src/ameba/rule/lint/spec_focus.cr:62]:
```suggestion
      return unless node.name.in?(SPEC_ITEM_NAMES)
```

File: src/ameba/rule/lint/spec_focus.cr [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      return if node.block.nil?
```
Comment (Sija) [inline src/ameba/rule/lint/spec_focus.cr:63]:
```suggestion
      return unless node.block
```

File: src/ameba/rule/lint/spec_focus.cr [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+      arg = node.named_args.try &.find { |a| a.name == "focus" }
```
Comment (Sija) [inline src/ameba/rule/lint/spec_focus.cr:65]:
```suggestion
      arg = node.named_args.try &.find(&.name.==("focus"))
```

**Review by Sija — COMMENTED** [review]
Couple of stylistic suggestions.

**Review by Sija — APPROVED** [review]
LGTM 🚀

File: src/ameba/source.cr [hunk 4]
```diff
@@ -53,7 +53,12 @@ module Ameba
       File.expand_path(path)
     end
 
-    # Returns true if *filepath* matches the source's path, false if it does not.
+    # Returns `true` if the source is a spec file, `false` if it is not.
+    def spec?
+      path.ends_with?("_spec.cr")
+    end
+
+    # Returns `true` if *filepath* matches the source's path, `false` if it does not.
```
Comment (Sija) [inline src/ameba/source.cr:61]:
```suggestion
    # Returns `true` if *filepath* matches the source's path, `false` otherwise.
```

**Comment by jwoertink** [comment]
Super stoked for this!

**Pull request merged**